rpc/rpccore/txn_cache.go

```diff
@@ -0,0 +1,107 @@
+package rpccore
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/NethermindEth/juno/core/felt"
+)
+
+const (
+	mapCapacityHint = 1024  // Assuming 1024 TPS
+	NumTimeBuckets  = 5 + 1 // TTL is 5s. 1 additional bucket for outdated entries.
+)
+
+type TxnCache struct {
```

Comment:
Some description here of what's the idea behind this. I am looking forward to understanding it